src/GLX/libglxmapping.c

```diff
@@ -447,6 +447,37 @@ static GLboolean LookupVendorEntrypoints(__GLXvendorInfo *vendor)
     return GL_TRUE;
 }
 
+static Bool CheckVendorImportsTable(__GLXvendorNameHash *pEntry)
+{
+    // Make sure all the required functions are there.
+    if (pEntry->imports.isScreenSupported == NULL
+            || pEntry->imports.getProcAddress == NULL
+            || pEntry->imports.getDispatchAddress == NULL
+            || pEntry->imports.setDispatchIndex == NULL)
+    {
+        return False;
+    }
+
+    // Check to see whether this vendor library can support entrypoint
+    // patching.
+    if (pEntry->imports.patchCallbacks != NULL) {
```

Comment:
I'm confused. Barring vendors overwriting the (const) pointer `patchCallbacks`, how can this be null ?

Fwiw I'm inclined to keep the following hunk in this function. It makes things more obvious

`pEntry->imports.patchCallbacks = &pEntry->patchCallbacks`